test/test_sort.py

```diff
@@ -514,6 +519,24 @@ def test_field_present_in_some_items(self):
         # items without field last
         assert [i.id for i in results_desc] == [ids[2], ids[1], ids[0], ids[3]]
 
+    def test_int_field_present_in_some_items(self):
+        """Test ordering by a field not present on all items."""
+        # append int-valued 'foo' to two items (1,2)
+        Item._types = {"foo": types.Integer()}
+        items = self.lib.items("id+")
+        ids = [i.id for i in items]
+        items[1].foo = 1
+        items[2].foo = 2
+        items[1].store()
+        items[2].store()
+
+        results_asc = list(self.lib.items("foo+ id+"))
+        # items without field first
+        assert [i.id for i in results_asc] == [ids[0], ids[3], ids[1], ids[2]]
+        results_desc = list(self.lib.items("foo- id+"))
+        # items without field last
+        assert [i.id for i in results_desc] == [ids[2], ids[1], ids[0], ids[3]]
+
```

Comment:
Consider naming variables in a slightly more clear way, something like this

```suggestion
    @patch("beets.library.Item._types", {"foo": types.Integer()})
    def test_int_field_present_in_some_items(self):
        """Test ordering by a field not present on all items."""
        # append int-valued 'foo' to two items (1,2)
        items = self.lib.items("id+")
        lower_foo_item, higher_foo_item, *items_without_foo = items
        lower_foo_item.foo = 1
        higher_foo_item.foo = 2
        lower_foo_item.store()
        higher_foo_item.store()

        items_without_foo_ids = [i.id for i in items_without_foo]
        results_asc = list(self.lib.items("foo+ id+"))
        # items without field first
        assert [i.id for i in results_asc] == [
            *items_without_foo_ids,
            lower_foo_item.id,
            higher_foo_item.id,
        ]
        results_desc = list(self.lib.items("foo- id+"))
        # items without field last
        assert [i.id for i in results_desc] == [
            higher_foo_item.id,
            lower_foo_item.id,
            *items_without_foo_ids,
        ]

```

Reply:
Done. Like the other change this one was c/p from elsewhere so I modified the source too.